back/backend/lib/common.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+import os
+
+from flask import request
+
+from backend.lib import formatters, auth, secrets
+from backend.models.region import Region
+from backend.models.province import Province
+from backend.models.user import Roles
+from backend.models.wca.event import Event
+from backend.models.wca.export import get_latest_export
+
+
+class Common(object):
+    current_date = datetime.datetime.now()
+
+    def __init__(self, wca_disclaimer=False):
+        self.uri = request.path
+        self.len = len
+        self.formatters = formatters
+        self.year = datetime.date.today().year
+        self.user = auth.user()
+        self.wca_disclaimer = wca_disclaimer
+
+        if self.user:
+            time_since_login = datetime.datetime.now() - self.user.last_login
+            if time_since_login < datetime.timedelta(seconds=1):
+                self.just_logged_in = True
+
+    def uri_matches(self, path):
+        return self.uri.endswith(path)
+
+    def uri_matches_any(self, path_list):
+        for text, path in path_list:
+            if self.uri_matches(path):
+                return True
+        return False
+
+    def wca_profile(self, wca_id):
+        return f"https://www.worldcubeassociation.org/persons/{wca_id}"
+
+    def format_date_range(self, start_date, end_date, include_year=True, full_months=False):
+        year_chunk = ', %d' % start_date.year if include_year else ''
+        month_format = lambda date: date.strftime('%B' if full_months else '%b')
```

Comment:
If we're assigning a lambda to a variable, then we should probably just be definining a normal function (although this is removed entirley by my comment above)